index.ts

```diff
@@ -25,12 +27,16 @@ app.get(
       level: Number(req.params.level),
     });
 
-    // TODO make this cached somehow so it doesn't get refetched every single time
-    // however, ensure that after sometime (maybe every hour or so) it _will_ fetch
-    // this fresh
-    const courses = await getCoursesFromSemester(coursesFromSemesterParams);
-
-    res.json(courses);
+    const id = req.originalUrl;
```

Comment:
imo you should construct the `id` from a serialized version of `coursesFromSemesterParams` so its not request dependent, like
```js
const id = `${coursesFromSemesterParams.year}-${coursesFromSemesterParams.term}-${coursesFromSemesterParams.level}`